CMakeLists.txt

```diff
@@ -143,7 +143,11 @@ if(NOT DEFINED SNMALLOC_ONLY_HEADER_LIBRARY)
   endmacro()
 
   macro(add_shim name)
-    add_library(${name} SHARED src/override/malloc.cc src/override/new.cc)
+    if(SNMALLOC_RUST_SUPPORT)
```

Comment:
This changes makes it hard to test the library on its own.  We really need to have something in CI that at least compiles your rust.cc code. And I would sooner not have an additional build machine spun up for that. 

Perhaps, we could have macro that takes a list of cc files to compile.  Then we can build both the Rust and the libC variants in the same build?

Reply:
will work on it ASAP.

Reply:
How about just change it to:
```
if(SNMALLOC_RUST_SUPPORT)
      add_library(${name}_rust STATIC src/override/rust.cc)
    endif()
    add_library(${name} SHARED src/override/malloc.cc src/override/new.cc)
```
Then you can set `SNMALLOC_RUST_SUPPORT=ON` at the pipelines, and I can change my linking targets.

Or, just as you have suggested, add a macro that takes a source file lists?

Reply:
You will then need to set all the relevant properties on `${name}_rust `, so you will end up with code duplication.